Crashes when run from root directory

> [ERROR] [SDL Error] Couldn't open /home/alxhoff/git/GitHub/resources
> "Failed to load image"    @-> /home/alxhoff/git/GitHub/FreeRTOS-Emulator/lib/Gfx/TUM_Draw.c:339, loadImage

--- a/lib/Gfx/TUM_Draw.c
+++ b/lib/Gfx/TUM_Draw.c
@@ -25,15 +25,12 @@
 #include <stdlib.h>
 #include <time.h>
 
-#include <assert.h>
-
 #include <SDL2/SDL.h>
 #include <SDL2/SDL2_gfxPrimitives.h>
 #include <SDL2/SDL_image.h>
 #include <SDL2/SDL_ttf.h>
 
 #include <pthread.h>
-#include <assert.h>
 
 #include "TUM_Draw.h"
 #include "TUM_Utils.h"
@@ -220,8 +217,8 @@ static draw_job_t *pushDrawJob(void)
 {
     draw_job_t *iterator;
     draw_job_t *job = calloc(1, sizeof(draw_job_t));
-
-    assert(job);
+    if(!job)
+        return NULL;
 
     for (iterator = &job_list_head; iterator->next;
          iterator = iterator->next)
@@ -345,18 +342,6 @@ static SDL_Texture *loadImage(char *filename, SDL_Renderer *ren)
 
     tex = IMG_LoadTexture(ren, filename);
 
-    if (!tex) {
-        if (ren) {
-            SDL_DestroyRenderer(ren);
-        }
-        if (window) {
-            SDL_DestroyWindow(window);
-        }
-        PRINT_SDL_ERROR("Failed to load image");
-        SDL_Quit();
-        exit(0);
-    }
-
     return tex;
 }
 
@@ -372,11 +357,16 @@ static void vRenderScaledImage(SDL_Texture *tex, SDL_Renderer *ren,
     SDL_RenderCopy(ren, tex, NULL, &dst);
 }
 
-void vGetImageSize(char *filename, int *w, int *h)
+static int vGetImageSize(char *filename, int *w, int *h)
 {
     SDL_Texture *tex = loadImage(filename, renderer);
+    if(!tex){
+        return -1;
+    }
     SDL_QueryTexture(tex, NULL, NULL, w, h);
     SDL_DestroyTexture(tex);
+
+    return 0;
 }
 
 static int vDrawScaledImage(SDL_Texture *tex, SDL_Renderer *ren,
@@ -384,6 +374,9 @@ static int vDrawScaledImage(SDL_Texture *tex, SDL_Renderer *ren,
 {
     int w, h;
     SDL_QueryTexture(tex, NULL, NULL, &w, &h);
+    if(!w || !h)
+        return -1;
+
     vRenderScaledImage(tex, ren, x, y, w * scale, h * scale);
     SDL_DestroyTexture(tex);
 
@@ -393,19 +386,21 @@ static int vDrawScaledImage(SDL_Texture *tex, SDL_Renderer *ren,
 static int vDrawImage(SDL_Texture *tex, SDL_Renderer *ren, signed short x,
                       signed short y)
 {
-    vDrawScaledImage(tex, ren, x, y, 1);
-
-    return 0;
+    return vDrawScaledImage(tex, ren, x, y, 1);
 }
 
-static void vDrawLoadAndDrawImage(char *filename, SDL_Renderer *ren,
+static int vDrawLoadAndDrawImage(char *filename, SDL_Renderer *ren,
                                   signed short x, signed short y)
 {
     SDL_Texture *tex = loadImage(filename, ren);
+    if(!tex)
+        return -1;
 
     vDrawImage(tex, ren, x, y);
 
     SDL_DestroyTexture(tex);
+
+    return 0;
 }
 
 static void vDrawRectImage(SDL_Texture *tex, SDL_Renderer *ren, SDL_Rect dst,
@@ -414,7 +409,7 @@ static void vDrawRectImage(SDL_Texture *tex, SDL_Renderer *ren, SDL_Rect dst,
     SDL_RenderCopy(ren, tex, clip, &dst);
 }
 
-static void vDrawClippedImage(SDL_Texture *tex, SDL_Renderer *ren,
+static int vDrawClippedImage(SDL_Texture *tex, SDL_Renderer *ren,
                               signed short x, signed short y,
                               SDL_Rect *clip)
 {
@@ -428,9 +423,13 @@ static void vDrawClippedImage(SDL_Texture *tex, SDL_Renderer *ren,
     }
     else {
         SDL_QueryTexture(tex, NULL, NULL, &dst.w, &dst.h);
+        if(!dst.x || !dst.y)
+            return -1;
     }
 
     vDrawRectImage(tex, ren, dst, clip);
+
+    return 0;
 }
 
 static int vDrawText(char *string, signed short x, signed short y,
@@ -452,16 +451,31 @@ static int vDrawText(char *string, signed short x, signed short y,
     return 0;
 }
 
-static void vGetTextSize(char *string, int *width, int *height)
+static int vGetTextSize(char *string, int *width, int *height)
 {
     SDL_Color color = { 0 };
     SDL_Surface *surface = TTF_RenderText_Solid(font, string, color);
-    assert(surface);
+    if(!surface)
+        goto err_surface;
+
     SDL_Texture *texture = SDL_CreateTextureFromSurface(renderer, surface);
-    assert(texture);
-    SDL_QueryTexture(texture, NULL, NULL, width, height);
+    if(!texture)
+        goto err_texture;
+
+    if(SDL_QueryTexture(texture, NULL, NULL, width, height))
+        goto err_query;
+
     SDL_DestroyTexture(texture);
     SDL_FreeSurface(surface);
+
+    return 0;
+
+err_query:
+    SDL_DestroyTexture(texture);
+err_texture:
+    SDL_FreeSurface(surface);
+err_surface:
+    return -1;
 }
 
 static int vDrawArrow(signed short x1, signed short y1, signed short x2,
@@ -510,7 +524,8 @@ static int vHandleDrawJob(draw_job_t *job)
         return -1;
     }
 
-    assert(job->data);
+    if(!job->data)
+        return -1;
 
     switch (job->type) {
         case DRAW_CLEAR:
@@ -591,6 +606,7 @@ static int vHandleDrawJob(draw_job_t *job)
 
 #define INIT_JOB(JOB, TYPE)                                                    \
     draw_job_t *JOB = pushDrawJob();                                       \
+    if(!JOB) return -1; \
     union data_u *data = calloc(1, sizeof(union data_u));                  \
     if (!data)                                                             \
         logCriticalError("job->data alloc");                           \
@@ -621,27 +637,28 @@ static float timespecDiffMilli(struct timespec *start, struct timespec *stop)
 #define FRAMELIMIT 60.0
 #define FRAMELIMIT_PERIOD 1000 / FRAMELIMIT
 
-void vDrawUpdateScreen(void)
+int vDrawUpdateScreen(void)
 {
     static struct timespec last_time = { 0 }, cur_time = { 0 };
 
     if (clock_gettime(CLOCK_MONOTONIC, &cur_time)) {
         PRINT_ERROR("Failed to get monotonic clock");
-        return;
+        goto err;
     }
 
     if (timespecDiffMilli(&last_time, &cur_time) < (float)FRAMELIMIT_PERIOD)
-        return;
+        goto err;
 
     memcpy(&last_time, &cur_time, sizeof(struct timespec));
 
     if (!job_list_head.next)
-        return;
+        goto err;
 
     draw_job_t *tmp_job;
 
     while ((tmp_job = popDrawJob()) != NULL) {
-        assert(tmp_job->data);
+        if(!tmp_job->data)
+            return -1;
         if (vHandleDrawJob(tmp_job) == -1) {
             goto draw_error;
         }
@@ -650,10 +667,12 @@ void vDrawUpdateScreen(void)
 
     SDL_RenderPresent(renderer);
 
-    return;
+    return 0;
 
 draw_error:
     free(tmp_job);
+err:
+    return -1;
 }
 
 char *tumGetErrorMessage(void)
@@ -797,7 +816,7 @@ void vExitDrawing(void)
     exit(EXIT_SUCCESS);
 }
 
-signed char tumDrawText(char *str, signed short x, signed short y,
+int tumDrawText(char *str, signed short x, signed short y,
                         unsigned int colour)
 {
     if (!strcmp(str, "")) {
@@ -810,7 +829,7 @@ signed char tumDrawText(char *str, signed short x, signed short y,
 
     if (!job->data->text.str) {
         printf("Error allocating buffer in tumDrawText\n");
-        exit(EXIT_FAILURE);
+        return -1;
     }
 
     strcpy(job->data->text.str, str);
@@ -821,13 +840,14 @@ signed char tumDrawText(char *str, signed short x, signed short y,
     return 0;
 }
 
-void tumGetTextSize(char *str, int *width, int *height)
+int tumGetTextSize(char *str, int *width, int *height)
 {
-    assert(str);
-    vGetTextSize(str, width, height);
+    if(!str)
+        return -1;
+    return vGetTextSize(str, width, height);
 }
 
-signed char tumDrawEllipse(signed short x, signed short y, signed short rx,
+int tumDrawEllipse(signed short x, signed short y, signed short rx,
                            signed short ry, unsigned int colour)
 {
     INIT_JOB(job, DRAW_ELLIPSE);
@@ -841,7 +861,7 @@ signed char tumDrawEllipse(signed short x, signed short y, signed short rx,
     return 0;
 }
 
-signed char tumDrawArc(signed short x, signed short y, signed short radius,
+int tumDrawArc(signed short x, signed short y, signed short radius,
                        signed short start, signed short end,
                        unsigned int colour)
 {
@@ -857,7 +877,7 @@ signed char tumDrawArc(signed short x, signed short y, signed short radius,
     return 0;
 }
 
-signed char tumDrawFilledBox(signed short x, signed short y, signed short w,
+int tumDrawFilledBox(signed short x, signed short y, signed short w,
                              signed short h, unsigned int colour)
 {
     INIT_JOB(job, DRAW_FILLED_RECT);
@@ -871,7 +891,7 @@ signed char tumDrawFilledBox(signed short x, signed short y, signed short w,
     return 0;
 }
 
-signed char tumDrawBox(signed short x, signed short y, signed short w,
+int tumDrawBox(signed short x, signed short y, signed short w,
                        signed short h, unsigned int colour)
 {
     INIT_JOB(job, DRAW_RECT);
@@ -885,7 +905,7 @@ signed char tumDrawBox(signed short x, signed short y, signed short w,
     return 0;
 }
 
-signed char tumDrawClear(unsigned int colour)
+int tumDrawClear(unsigned int colour)
 {
     /** INIT_JOB(job, DRAW_CLEAR); */
     draw_job_t *job = pushDrawJob();
@@ -901,7 +921,7 @@ signed char tumDrawClear(unsigned int colour)
     return 0;
 }
 
-signed char tumDrawCircle(signed short x, signed short y, signed short radius,
+int tumDrawCircle(signed short x, signed short y, signed short radius,
                           unsigned int colour)
 {
     INIT_JOB(job, DRAW_CIRCLE);
@@ -914,7 +934,7 @@ signed char tumDrawCircle(signed short x, signed short y, signed short radius,
     return 0;
 }
 
-signed char tumDrawLine(signed short x1, signed short y1, signed short x2,
+int tumDrawLine(signed short x1, signed short y1, signed short x2,
                         signed short y2, unsigned char thickness,
                         unsigned int colour)
 {
@@ -930,7 +950,7 @@ signed char tumDrawLine(signed short x1, signed short y1, signed short x2,
     return 0;
 }
 
-signed char tumDrawPoly(coord_t *points, int n, unsigned int colour)
+int tumDrawPoly(coord_t *points, int n, unsigned int colour)
 {
     INIT_JOB(job, DRAW_POLY);
 
@@ -941,7 +961,7 @@ signed char tumDrawPoly(coord_t *points, int n, unsigned int colour)
     return 0;
 }
 
-signed char tumDrawTriangle(coord_t *points, unsigned int colour)
+int tumDrawTriangle(coord_t *points, unsigned int colour)
 {
     INIT_JOB(job, DRAW_TRIANGLE);
 
@@ -951,7 +971,7 @@ signed char tumDrawTriangle(coord_t *points, unsigned int colour)
     return 0;
 }
 
-signed char tumDrawImage(char *filename, signed short x, signed short y)
+int tumDrawImage(char *filename, signed short x, signed short y)
 {
     INIT_JOB(job, DRAW_IMAGE);
 
@@ -970,18 +990,17 @@ signed char tumDrawImage(char *filename, signed short x, signed short y)
     return 0;
 }
 
-void tumGetImageSize(char *filename, int *w, int *h)
+int tumGetImageSize(char *filename, int *w, int *h)
 {
     char full_filename[PATH_MAX + 1];
     realpath(filename, full_filename);
-    vGetImageSize(full_filename, w, h);
+    return vGetImageSize(full_filename, w, h);
 }
 
-signed char tumDrawScaledImage(char *filename, signed short x, signed short y,
+int tumDrawScaledImage(char *filename, signed short x, signed short y,
                                float scale)
 {
     // TODO
-
     INIT_JOB(job, DRAW_SCALED_IMAGE);
 
     char abs_path[PATH_MAX + 1];
@@ -1000,7 +1019,7 @@ signed char tumDrawScaledImage(char *filename, signed short x, signed short y,
     return 0;
 }
 
-signed char tumDrawArrow(signed short x1, signed short y1,
+int tumDrawArrow(signed short x1, signed short y1,
                          signed short x2, signed short y2,
                          signed short head_length, unsigned char thickness,
                          unsigned int colour)

--- a/lib/Gfx/include/TUM_Draw.h
+++ b/lib/Gfx/include/TUM_Draw.h
@@ -145,17 +145,17 @@ void vExitDrawing(void);
  * each function queues a draw job into a queue. Once vDrawUpdateScreen is called,
  * the queued draw jobs are executed by the background SDL thread.
  *
- * @returns NULL always return NULL
+ * @returns 0 on success
  */
-void vDrawUpdateScreen(void);
+int vDrawUpdateScreen(void);
 
 /**
  * @brief Sets the screen to a solid colour
  *
  * @param colour RGB colour to fill the screen with
- * @return signed char 0 on success
+ * @return 0 on success
  */
-signed char tumDrawClear(unsigned int colour);
+int tumDrawClear(unsigned int colour);
 
 /**
  * @brief Draws an ellipse on the screen
@@ -165,10 +165,10 @@ signed char tumDrawClear(unsigned int colour);
  * @param rx Horizontal radius in pixels
  * @param ry Vertical radius in pixels
  * @param colour RGB colour of the ellipse
- * @return signed char 0 on success
+ * @return 0 on success
  */
-signed char tumDrawEllipse(signed short x, signed short y, signed short rx,
-                           signed short ry, unsigned int colour);
+int tumDrawEllipse(signed short x, signed short y, signed short rx,
+                   signed short ry, unsigned int colour);
 
 /**
  * @brief Draws an arc on the screen
@@ -182,11 +182,11 @@ signed char tumDrawEllipse(signed short x, signed short y, signed short rx,
  * @param start Starting radius of the arc, 0 degrees is down
  * @param end Ending radius of the arc, 0 degrees is down
  * @param colour RGB colour of the arc
- * @return signed char 0 on success
+ * @return 0 on success
  */
-signed char tumDrawArc(signed short x, signed short y, signed short radius,
-                       signed short start, signed short end,
-                       unsigned int colour);
+int tumDrawArc(signed short x, signed short y, signed short radius,
+               signed short start, signed short end,
+               unsigned int colour);
 
 /**
  * @brief Prints a string to the screen
@@ -198,10 +198,10 @@ signed char tumDrawArc(signed short x, signed short y, signed short radius,
  * @param x X coordinate of the top left point of the text's bounding box
  * @param y Y coordinate of the top left point of the text's bounding box
  * @param colour RGB colour of the text
- * @return signed char 0 on success
+ * @return 0 on success
  */
-signed char tumDrawText(char *str, signed short x, signed short y,
-                        unsigned int colour);
+int tumDrawText(char *str, signed short x, signed short y,
+                unsigned int colour);
 
 /**
  * @brief Finds the width and height of a strings bounding box
@@ -209,9 +209,9 @@ signed char tumDrawText(char *str, signed short x, signed short y,
  * @param str String who's bounding box size is required
  * @param width Integer where the width shall be stored
  * @param height Integer where the height shall be stored
- * @return signed char 0 on success
+ * @return 0 on success
  */
-void tumGetTextSize(char *str, int *width, int *height);
+int tumGetTextSize(char *str, int *width, int *height);
 
 /**
  * @brief Draws a filled box on the screen
@@ -221,10 +221,10 @@ void tumGetTextSize(char *str, int *width, int *height);
  * @param w Width of the box
  * @param h Height of the box
  * @param colour RGB colour of the box
- * @return signed char 0 on success
+ * @return 0 on success
  */
-signed char tumDrawBox(signed short x, signed short y, signed short w,
-                       signed short h, unsigned int colour);
+int tumDrawBox(signed short x, signed short y, signed short w,
+               signed short h, unsigned int colour);
 
 /**
  * @brief Draws an unfilled box on the screen
@@ -234,10 +234,10 @@ signed char tumDrawBox(signed short x, signed short y, signed short w,
  * @param w Width of the box
  * @param h Height of the box
  * @param colour RGB colour of the filled box
- * @return signed char 0 on success
+ * @return 0 on success
  */
-signed char tumDrawFilledBox(signed short x, signed short y, signed short w,
-                             signed short h, unsigned int colour);
+int tumDrawFilledBox(signed short x, signed short y, signed short w,
+                     signed short h, unsigned int colour);
 
 /**
  * @brief Draws a filled circle on the screen
@@ -246,10 +246,10 @@ signed char tumDrawFilledBox(signed short x, signed short y, signed short w,
  * @param y Y coordinate of the center of the circle
  * @param radius Radius of the circle in pixels
  * @param colour RGB colour of the ellipse
- * @return signed char 0 on success
+ * @return 0 on success
  */
-signed char tumDrawCircle(signed short x, signed short y, signed short radius,
-                          unsigned int colour);
+int tumDrawCircle(signed short x, signed short y, signed short radius,
+                  unsigned int colour);
 
 /**
  * @brief Draws a line on the screen
@@ -260,11 +260,11 @@ signed char tumDrawCircle(signed short x, signed short y, signed short radius,
  * @param y2 x coordinate of the starting point of the line
  * @param thickness The thickness of the line in pixels
  * @param colour RGB colour of the ellipse
- * @return signed char 0 on success
+ * @return 0 on success
  */
-signed char tumDrawLine(signed short x1, signed short y1, signed short x2,
-                        signed short y2, unsigned char thickness,
-                        unsigned int colour);
+int tumDrawLine(signed short x1, signed short y1, signed short x2,
+                signed short y2, unsigned char thickness,
+                unsigned int colour);
 
 /**
  * @brief Draws a polygon on the screen
@@ -276,18 +276,18 @@ signed char tumDrawLine(signed short x1, signed short y1, signed short x2,
  * @param points Points array specifying each point in the polygon
  * @param n Number of points in the points array
  * @param colour RGB colour of the ellipse
- * @return signed char 0 on success
+ * @return 0 on success
  */
-signed char tumDrawPoly(coord_t *points, int n, unsigned int colour);
+int tumDrawPoly(coord_t *points, int n, unsigned int colour);
 
 /**
  * @brief Draws a triangle on the screen
  *
  * @param points Points array giving the three corner points of the triangle
  * @param colour RGB colour of the ellipse
- * @return signed char 0 on success
+ * @return 0 on success
  */
-signed char tumDrawTriangle(coord_t *points, unsigned int colour);
+int tumDrawTriangle(coord_t *points, unsigned int colour);
 
 /**
  * @brief Draws an image on the screen
@@ -295,9 +295,9 @@ signed char tumDrawTriangle(coord_t *points, unsigned int colour);
  * @param filename Filename of the image to be drawn
  * @param x X coordinate of the top left corner of the image
  * @param y Y coordinate of the top left corner of the image
- * @return signed char 0 on success
+ * @return 0 on success
  */
-signed char tumDrawImage(char *filename, signed short x, signed short y);
+int tumDrawImage(char *filename, signed short x, signed short y);
 
 /**
  * @brief Gets the width and height of an image
@@ -305,9 +305,9 @@ signed char tumDrawImage(char *filename, signed short x, signed short y);
  * @param filename Image filename to be tested
  * @param w Integer where the width shall be stored
  * @param h Integer where the height shall be stored
- * @return NULL always returns NULL
+ * @return 0 on sucess
  */
-void tumGetImageSize(char *filename, int *w, int *h);
+int tumGetImageSize(char *filename, int *w, int *h);
 
 /**
  * @brief Draws a scaled image on the screen
@@ -316,10 +316,10 @@ void tumGetImageSize(char *filename, int *w, int *h);
  * @param x X coordinate of the top left corner of the image
  * @param y Y coordinate of the top left corner of the image
  * @param scale The scale factor of the image
- * @return signed char 0 on success
+ * @return 0 on success
  */
-signed char tumDrawScaledImage(char *filename, signed short x, signed short y,
-                               float scale);
+int tumDrawScaledImage(char *filename, signed short x, signed short y,
+                       float scale);
 
 /**
  * @brief Draws an arrow on the screen
@@ -331,12 +331,12 @@ signed char tumDrawScaledImage(char *filename, signed short x, signed short y,
  * @param head_length Length in pixels of the arrow's head
  * @param thickness Thickness in pixels of the arrow's lines
  * @param colour RGB colour of the ellipse
- * @return signed char 0 on success
+ * @return 0 on success
  */
-signed char tumDrawArrow(signed short x1, signed short y1,
-                         signed short x2, signed short y2,
-                         signed short head_length, unsigned char thickness,
-                         unsigned int colour);
+int tumDrawArrow(signed short x1, signed short y1,
+                 signed short x2, signed short y2,
+                 signed short head_length, unsigned char thickness,
+                 unsigned int colour);
 
 /**
  * @}

--- a/src/main.c
+++ b/src/main.c
@@ -1,7 +1,6 @@
 #include <math.h>
 #include <stdio.h>
 #include <stdlib.h>
-#include <assert.h>
 
 #include <SDL2/SDL_scancode.h>
 
@@ -94,8 +93,6 @@ void checkDraw(unsigned char status, const char *msg)
         else {
             fprintf(stderr, "[ERROR] %s\n", tumGetErrorMessage());
         }
-
-        exit(EXIT_FAILURE);
     }
 }
 
@@ -231,7 +228,8 @@ void vDrawCave(void)
     circlePositionX = CAVE_X + xGetMouseX() / 2;
     circlePositionY = CAVE_Y + xGetMouseY() / 2;
 
-    tumDrawCircle(circlePositionX, circlePositionY, 20, Green);
+    checkDraw(tumDrawCircle(circlePositionX, circlePositionY, 20, Green),
+              __FUNCTION__);
 }
 
 void vDrawHelpText(void)
@@ -241,18 +239,17 @@ void vDrawHelpText(void)
 
     sprintf(str, "[Q]uit, [C]hange State");
 
-    tumGetTextSize((char *)str, &text_width, NULL);
-
-    checkDraw(tumDrawText(str, SCREEN_WIDTH - text_width - 10,
-                          DEFAULT_FONT_SIZE * 0.5, Black),
-              __FUNCTION__);
+    if (!tumGetTextSize((char *)str, &text_width, NULL))
+        checkDraw(tumDrawText(str, SCREEN_WIDTH - text_width - 10,
+                              DEFAULT_FONT_SIZE * 0.5, Black),
+                  __FUNCTION__);
 }
 
 #define FPS_AVERAGE_COUNT 50
 
 void vDrawFPS(void)
 {
-    static unsigned int periods[FPS_AVERAGE_COUNT] = {0};
+    static unsigned int periods[FPS_AVERAGE_COUNT] = { 0 };
     static unsigned int periods_total = 0;
     static unsigned int index = 0;
     static unsigned int average_count = 0;
@@ -264,7 +261,8 @@ void vDrawFPS(void)
     xLastWakeTime = xTaskGetTickCount();
 
     if (prevWakeTime != xLastWakeTime) {
-        periods[index] = configTICK_RATE_HZ / (xLastWakeTime - prevWakeTime);
+        periods[index] =
+            configTICK_RATE_HZ / (xLastWakeTime - prevWakeTime);
         prevWakeTime = xLastWakeTime;
     }
     else {
@@ -291,21 +289,24 @@ void vDrawFPS(void)
 
     sprintf(str, "FPS: %2d", fps);
 
-    tumGetTextSize((char *)str, &text_width, NULL);
-
-    checkDraw(tumDrawText(str, SCREEN_WIDTH - text_width - 10,
-                          SCREEN_HEIGHT - DEFAULT_FONT_SIZE * 1.5, Blue),
-              __FUNCTION__);
+    if (!tumGetTextSize((char *)str, &text_width, NULL))
+        checkDraw(tumDrawText(str, SCREEN_WIDTH - text_width - 10,
+                              SCREEN_HEIGHT - DEFAULT_FONT_SIZE * 1.5,
+                              Blue),
+                  __FUNCTION__);
 }
 
 void vDrawLogo(void)
 {
     static int image_height;
-    tumGetImageSize(LOGO_FILENAME, NULL, &image_height);
-    checkDraw(tumDrawScaledImage(LOGO_FILENAME, 10,
-                                 SCREEN_HEIGHT - 10 - image_height * 0.3,
-                                 0.3),
-              __FUNCTION__);
+    if (!tumGetImageSize(LOGO_FILENAME, NULL, &image_height))
+        checkDraw(tumDrawScaledImage(
+                      LOGO_FILENAME, 10,
+                      SCREEN_HEIGHT - 10 - image_height * 0.3, 0.3),
+                  __FUNCTION__);
+    else {
+        fprintf(stderr, "Failed to get size of image '%s', does it exist?\n", LOGO_FILENAME);
+    }
 }
 
 void vDrawStaticItems(void)
@@ -424,15 +425,14 @@ void vDemoSendTask(void *pvParameters)
         }
 
         if (udp_soc_one)
-            assert(!aIOSocketPut(UDP, NULL, UDP_TEST_PORT_1,
-                                 test_str_1, strlen(test_str_1)));
+            aIOSocketPut(UDP, NULL, UDP_TEST_PORT_1,
+                         test_str_1, strlen(test_str_1));
         if (udp_soc_two)
-            assert(!aIOSocketPut(UDP, NULL, UDP_TEST_PORT_2,
-                                 test_str_2, strlen(test_str_2)));
+            aIOSocketPut(UDP, NULL, UDP_TEST_PORT_2,
+                         test_str_2, strlen(test_str_2));
         if (tcp_soc)
-            assert(!aIOSocketPut(TCP, NULL, TCP_TEST_PORT,
-                                 test_str_3, strlen(test_str_3)));
-
+            aIOSocketPut(TCP, NULL, TCP_TEST_PORT,
+                         test_str_3, strlen(test_str_3));
 
         vTaskDelay(pdMS_TO_TICKS(1000));
     }
@@ -640,7 +640,6 @@ int main(int argc, char *argv[])
         PRINT_ERROR("Failed to create buttons lock");
         goto err_buttons_lock;
     }
-    assert(buttons.lock);
 
     DrawSignal = xSemaphoreCreateBinary(); // Screen buffer locking
     if (!DrawSignal) {
@@ -661,25 +660,26 @@ int main(int argc, char *argv[])
     }
 
     if (xTaskCreate(basicSequentialStateMachine, "StateMachine",
-                    mainGENERIC_STACK_SIZE * 2, NULL, configMAX_PRIORITIES - 1,
-                    StateMachine) != pdPASS) {
+                    mainGENERIC_STACK_SIZE * 2, NULL,
+                    configMAX_PRIORITIES - 1, StateMachine) != pdPASS) {
         PRINT_TASK_ERROR("StateMachine");
         goto err_statemachine;
     }
-    if (xTaskCreate(vSwapBuffers, "BufferSwapTask", mainGENERIC_STACK_SIZE * 2,
-                    NULL, configMAX_PRIORITIES, BufferSwap) != pdPASS) {
+    if (xTaskCreate(vSwapBuffers, "BufferSwapTask",
+                    mainGENERIC_STACK_SIZE * 2, NULL, configMAX_PRIORITIES,
+                    BufferSwap) != pdPASS) {
         PRINT_TASK_ERROR("BufferSwapTask");
         goto err_bufferswap;
     }
 
     /** Demo Tasks */
-    if (xTaskCreate(vDemoTask1, "DemoTask1", mainGENERIC_STACK_SIZE * 2, NULL,
-                    mainGENERIC_PRIORITY, &DemoTask1) != pdPASS) {
+    if (xTaskCreate(vDemoTask1, "DemoTask1", mainGENERIC_STACK_SIZE * 2,
+                    NULL, mainGENERIC_PRIORITY, &DemoTask1) != pdPASS) {
         PRINT_TASK_ERROR("DemoTask1");
         goto err_demotask1;
     }
-    if (xTaskCreate(vDemoTask2, "DemoTask2", mainGENERIC_STACK_SIZE * 2, NULL,
-                    mainGENERIC_PRIORITY, &DemoTask2) != pdPASS) {
+    if (xTaskCreate(vDemoTask2, "DemoTask2", mainGENERIC_STACK_SIZE * 2,
+                    NULL, mainGENERIC_PRIORITY, &DemoTask2) != pdPASS) {
         PRINT_TASK_ERROR("DemoTask2");
         goto err_demotask2;
     }

